src/Administration/Resources/app/administration/src/app/filter/currency.filter.js

```diff
@@ -1,8 +1,9 @@
 const { Filter } = Shopware;
+const { types } = Shopware.Utils;
 const { currency } = Shopware.Utils.format;
 
 Filter.register('currency', (value, format, decimalPlaces) => {
-    if (value === null) {
+    if (!value && !types.isNumber(value)) {
```

Comment:
Different behavior. Use `if (value === null || value === undefined)`

Reply:
I think the given way in my PR is the better way. On this way all falsy values are handled appropriately and numbers are still passed to the calculation. Falsy values should never be able to cause issues, no matter if it's `null`, `undefined`, an empty string or maybe even different values I can't think of right now.

Reply:
There are plenty values which are falsy. Here are some examples:

- the number 0
- the BigInt 0n
- the keyword null
- the keyword undefined
- the boolean false
- the number NaN
- the empty string "" (equivalent to '' or ``)

What is when the value is 0? Then it returns `-` instead of the real value. Therefore I would prefer explicit checks

Reply:
no, it does not return `-`, because 0 is covered by the `!types.isNumber(value)`. The same counts for NaN. They both are numbers so they do have a `.toLocaleString()` function, which is later used in this context. 

The same goes for the BigInt, null, undefined, false and an empty string. All values are caught by both `!value` and `!types.isNumber(value)`.
Lets check:
0 -> true && false (because it is a number)
0n -> true && false (because it is a number)
null -> true && true (because it's not a number)
undefined -> true && true (because it's not a number)
false -> true && true (because it's not a number)
NaN -> true && false (because it is a number)
empty string -> true && true (because it's not a number)

If you'd like to test this specific example I'm open on planning a zoom call to test this out :)

Reply:
My fault. I overlooked the `!types.isNumber(value)`. Then its fine

Reply:
a small test for all these cases and the discussion would be obsolete :smile: